### PR TITLE
Migrate `firebase-appcheck` to use standard executors provided by Firebase Common.

### PR DESCRIPTION
**Pull request by rosalyntan**
App Check executor migration:

- [x] `firebase-appcheck`
- [ ] `firebase-appcheck-safetynet`
- [ ] `firebase-appcheck-playintegrity`
- [ ] `firebase-appcheck-debug`

### CODE REVIEW

**Review by vkryachko — COMMENTED** [review]
can you also update tests to use `TestOnlyExecutors` where appropriate, this will make sure any disallowed io is not performed on the executors that you picked for appcheck?

File: appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultFirebaseAppCheck.java [hunk 1]
```diff
@@ -50,29 +52,19 @@ public class DefaultFirebaseAppCheck extends FirebaseAppCheck {
   private final StorageHelper storageHelper;
   private final TokenRefreshManager tokenRefreshManager;
   private final ExecutorService backgroundExecutor;
```
[the remark this answers is not in the document]
Comment (rosalyntan) [reply appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultFirebaseAppCheck.java]: Done.

File: appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultTokenRefresher.java [hunk 2]
```diff
@@ -41,10 +40,8 @@ public class DefaultTokenRefresher {
   private volatile ScheduledFuture<?> refreshFuture;
   private volatile long delayAfterFailureSeconds;
 
-  // TODO(b/258273630): Migrate to go/firebase-android-executors
-  @SuppressLint("ThreadPoolCreation")
   DefaultTokenRefresher(@NonNull DefaultFirebaseAppCheck firebaseAppCheck) {
-    this(checkNotNull(firebaseAppCheck), Executors.newScheduledThreadPool(/* corePoolSize= */ 1));
+    this(checkNotNull(firebaseAppCheck), firebaseAppCheck.getScheduledExecutorService());
```
[the remark this answers is not in the document]
Comment (rosalyntan) [reply appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultTokenRefresher.java]: Done.

**Comment by rosalyntan** [comment]
> can you also update tests to use TestOnlyExecutors where appropriate, this will make sure any disallowed io is not performed on the executors that you picked for appcheck?

Chatted offline -- using a real background executor instead of `MoreExecutors.directExecutor()` causes tests to fail because asserts run before background tasks have finished. Added a TODO to refactor the tests in the future.

**Review by vkryachko — APPROVED** [review]

**Pull request merged**